From 71f890dc6036d87ec114d2846aa8c21e55466d4d Mon Sep 17 00:00:00 2001
Message-Id: <patch-01.13-71f890dc60-20210523T085231Z-avarab@gmail.com>
In-Reply-To: <cover-00.13-0000000000-20210523T085231Z-avarab@gmail.com>
References: <cover-00.10-00000000000-20210520T081826Z-avarab@gmail.com>
	<cover-00.13-0000000000-20210523T085231Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Sun, 23 May 2021 10:41:37 +0200
Subject: [PATCH v3 01/13] send-email tests: support
 GIT_TEST_PERL_FATAL_WARNINGS=true
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Add support for the "GIT_TEST_PERL_FATAL_WARNINGS=true" test mode to
"send-email". This was added to e.g. git-svn in 5338ed2b26 (perl:
check for perl warnings while running tests, 2020-10-21), but not
"send-email". Let's rectify that.

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 git-send-email.perl | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/git-send-email.perl b/git-send-email.perl
index 175da07d94..cffdfdacfb 100755
--- a/git-send-email.perl
+++ b/git-send-email.perl
@@ -18,7 +18,7 @@
 
 use 5.008;
 use strict;
-use warnings;
+use warnings $ENV{GIT_PERL_FATAL_WARNINGS} ? qw(FATAL all) : ();
 use POSIX qw/strftime/;
 use Term::ReadLine;
 use Getopt::Long;
-- 
2.32.0.rc0.406.g05cb3eebfc

